frontend/src/pages/TermPlanner/SettingsMenu/SettingsMenu.tsx

```diff
@@ -23,35 +21,83 @@ const SettingsMenu = ({ planner }: Props) => {
   const { Option } = Select;
   const { token, theme } = useSelector((state: RootState) => state.settings);
 
-  async function handleUpdateStartYear(_: unknown, dateString: string | string[]) {
-    if (dateString && typeof dateString === 'string') {
-      try {
-        await axios.put(
-          '/user/updateStartYear',
-          { startYear: parseInt(dateString, 10) },
-          { params: { token } }
-        );
-      } catch {
-        openNotification({
-          type: 'error',
-          message: 'Error setting degree start year',
-          description: 'There was an error updating the degree start year.'
-        });
+  function willUnplanCourses(numYears: number) {
+    if (!planner) return false;
+
+    if (numYears < planner.years.length) {
+      for (let i = numYears; i < planner.years.length; i++) {
+        if (Object.values(planner.years[i]).flat().length > 0) return true;
       }
     }
+
+    return false;
   }
 
-  async function handleUpdateDegreeLength(value: number) {
-    try {
-      await axios.put('/user/updateDegreeLength', { numYears: value }, { params: { token } });
-    } catch {
+  async function updateStartYear(year: string) {
+    await axios.put(
+      '/user/updateStartYear',
+      { startYear: parseInt(year, 10) },
+      { params: { token } }
+    );
+  }
```

Comment:
(in future all api requests will need to be put in helpers inside the api folder, but this is ok right now since i know its done a lot of places)